File: Adorable/app/src/main/java/merryweather/com/adorable/di/component/AppComponent.java
package merryweather.com.adorable.di.component;

import android.content.Context;

import javax.inject.Singleton;

import dagger.Component;
import merryweather.com.adorable.App;
import merryweather.com.adorable.api.OpenWeatherMapService;
import merryweather.com.adorable.di.modules.AppModule;
import merryweather.com.adorable.di.modules.PopViewersModule;
import merryweather.com.adorable.di.modules.RepositoryModule;
import merryweather.com.adorable.di.modules.RetrofitModule;
import merryweather.com.adorable.di.qualifier.AppContext;

/**
 * Created by S on 19.04.2018.
 */

@Singleton
@Component(modules = {AppModule.class, PopViewersModule.class, RetrofitModule.class, RepositoryModule.class})
public interface AppComponent {

    @AppContext
    Context context();

    OpenWeatherMapService openWeatherMapService();

    void inject(final App theApp);
}

File: Tech45/app/src/main/java/merryweather/com/tech45/di/component/AppComponent.kt
package merryweather.com.tech45.di.component

import android.content.Context

import javax.inject.Singleton

import dagger.Component
import merryweather.com.tech45.App
import merryweather.com.tech45.api.TechService
import merryweather.com.tech45.di.modules.AppModule
import merryweather.com.tech45.di.modules.RepositoryModule
import merryweather.com.tech45.di.modules.RetrofitModule
import merryweather.com.tech45.di.modules.TechModule
import merryweather.com.tech45.di.qualifier.AppContext

/**
 * Created by S on 19.04.2018.
 */

@Singleton
@Component(modules = arrayOf(AppModule::class, TechModule::class, RetrofitModule::class, RepositoryModule::class))
interface AppComponent {

    @AppContext
    fun context(): Context

    fun ltechService(): TechService


    fun inject(theApp: App)
}
